modules/social_features/social_group/src/Form/SocialGroupSettings.php

```diff
@@ -95,8 +201,19 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
       $config->set($key, !empty($value));
     }
 
-    $config->set('default_hero', $form_state->getValue('default_hero'))->save();
-    $config->set('social_group_type_required', $form_state->getValue('social_group_type_required'))->save();
+    $cross_posting_status = $form_state->getValue(['cross_posting', 'status']);
+    $config->set('cross_posting.status', $cross_posting_status);
+    $config->set('cross_posting.content_types', $cross_posting_status
+      ? Checkboxes::getCheckedCheckboxes($form_state->getValue(['cross_posting', 'content_types']))
+      : []
+    );
+    $config->set('cross_posting.group_types_disabled', $cross_posting_status
```

Comment:
To be consistent and inclusive I would propose use just group_types variable name, title and logic in all places.